easypy/signals.py

```diff
@@ -146,11 +158,15 @@ class Signal:
 
     ALL = {}
 
-    def __new__(cls, name, asynchronous=None, swallow_exceptions=False, log=True):
+    def __new__(cls, name, asynchronous=None, swallow_exceptions=False, log=True, signature=None):
         try:
-            return cls.ALL[name]
+            signal = cls.ALL[name]
```

Comment:
I think we don't need try/except here. `if name in cls.ALL` is more elegant.

Reply:
This is the superior style. `try:` clause to get the signal, `except KeyError:` clause the handle the case it does not exist, `else:` clause to handle the case it does exist.